dotCMS/src/integration-test/java/com/dotcms/publisher/util/DependencyManagerTest.java

```diff
@@ -0,0 +1,256 @@
+import com.dotcms.publisher.pusher.PushPublisherConfig;
+import com.dotcms.publishing.DotBundleException;
+import com.dotcms.publishing.PublisherConfig.Operation;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.RelationshipAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.structure.model.ContentletRelationships;
+import com.dotmarketing.portlets.structure.model.ContentletRelationships.ContentletRelationshipRecords;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.util.WebKeys.Relationship.RELATIONSHIP_CARDINALITY;
+import com.google.common.collect.Lists;
+import com.liferay.portal.model.User;
+import java.util.Date;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author nollymar
+ */
+public class DependencyManagerTest {
+
+    private static long languageId;
+    private static User user;
+    private static BundleAPI bundleAPI;
+    private static ContentletAPI contentletAPI;
+    private static FieldAPI contentTypeFieldAPI;
+    private static RelationshipAPI relationshipAPI;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        languageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        user = APILocator.systemUser();
+
+        bundleAPI = APILocator.getBundleAPI();
+        contentletAPI = APILocator.getContentletAPI();
+        contentTypeFieldAPI = APILocator.getContentTypeFieldAPI();
+        relationshipAPI = APILocator.getRelationshipAPI();
+    }
+
+    /**
+     * <b>Method to test:</b> {@link DependencyManager#setDependencies()} <p>
+     * <b>Given Scenario:</b> A bundle contains a content with a self-join relationship <p>
+     * <b>ExpectedResult:</b> The Dependency Manager should include all related content to the bundle
+     */
+    @Test
+    public void test_dependencyManager_shouldIncludeSelfRelationships()
+            throws DotSecurityException, DotBundleException, DotDataException {
+
+        final PushPublisherConfig config = new PushPublisherConfig();
+        final ContentType contentType = getContentTypeWithSelfJoinRelationship();
+        final ContentletDataGen dataGen = new ContentletDataGen(contentType.id());
+
+        //Creates parent content
+        Contentlet blogContentParent = dataGen
+                .languageId(languageId)
+                .setProperty("title", "blogContentParent")
+                .setProperty("urlTitle", "blogContentParent")
+                .setProperty("author", "systemUser")
+                .setProperty("sysPublishDate", new Date())
```

Comment:
Codacy found an issue: [The String literal "sysPublishDate" appears 4 times in this file; the first occurrence is on line 83](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5775242)